### PR TITLE
Unify docker-compose files with sensible defaults

### PR DESCRIPTION
**Pull request by nwalters512**
This drops the separate `docker-compose-production.yml` file in favor of a single `docker-compose.yml` file that can work well for both production-like usage and local development.

It defaults to production-like usage so that e.g. course staff can easily use Docker Compose to start up a local instance for developing course content. There are also a few commented-out bits of config that can make it work better for developing PL itself (enabling Nodemon and mounting in a local copy of PL source).

### CODE REVIEW

File: docs/docker.md [hunk 1]
```diff
@@ -71,47 +71,27 @@ E.g., to start a shell in a container started with `--name pl`:
...
 
 This section describes common applications for [Docker Compose](https://github.com/docker/compose) with PrairieLearn. See the [official Docker Compose documentation](https://docs.docker.com/compose/) for more.
 
 ### Basics
 
-A docker-compose file describes the services an application needs to run. In our case, we use `docker-compose` to configure and run the PrairieLearn docker container locally.
+A docker-compose file describes the services an application needs to run. In our case, we use `docker compose` to configure and run the PrairieLearn docker container locally.
 
-To run PrairieLearn with `docker-compose`, run `docker-compose up pl`. This will, in order:
+To run PrairieLearn with `docker compose`, run `docker compose up`.
```
Comment (mwest1066) [inline docs/docker.md:82]: At least for me the `up` was required here.

Comment (mwest1066) [reply docs/docker.md:82]: I ran `yarn`, pulled a new `prairielearn/prairielearn` image, and then ran `docker compose up`. It failed with:
```
pl  | node:internal/modules/cjs/loader:936
pl  |   throw err;
pl  |   ^
pl  |
pl  | Error: Cannot find module '@prairielearn/sentry'
pl  | Require stack:
pl  | - /PrairieLearn/workspace_host/interface.js
pl  |     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
pl  |     at Function.Module._load (node:internal/modules/cjs/loader:778:27)
pl  |     at Module.require (node:internal/modules/cjs/loader:1005:19)
pl  |     at require (node:internal/modules/cjs/helpers:102:18)
pl  |     at Object.<anonymous> (/PrairieLearn/workspace_host/interface.js:23:16)
pl  |     at Module._compile (node:internal/modules/cjs/loader:1105:14)
pl  |     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
pl  |     at Module.load (node:internal/modules/cjs/loader:981:32)
pl  |     at Function.Module._load (node:internal/modules/cjs/loader:822:12)
pl  |     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
pl  |     at node:internal/main/run_main_module:17:47 {
pl  |   code: 'MODULE_NOT_FOUND',
pl  |   requireStack: [ '/PrairieLearn/workspace_host/interface.js' ]
pl  | }
```

**Review by mwest1066 — APPROVED** [review]

File: docs/docker.md (hunk 1 above)
Comment (nwalters512) [reply docs/docker.md:82]: Hmm, is that with our without your repo mounted into `/PrairieLearn`?

Comment (nwalters512) [reply docs/docker.md:82]: That is very strange, as the exact same sequence yields a running image for me 🙃

**Pull request closed without merging**